File: Cpp/Armstrong Number.cpp
   # include <iostream>
   # include "conio.h"
   # include "math.h"
   using namespace std;
   int main ()
   {
     //clrscr();
     int a,b=0,sum=0;
     long int n;
     cout<<"Enter the NO. : ";
     cin>>n;
     for(;n>0;)
     //counts the digits
     {
       a=n%10;
       n=n/10;
       b++;
     }
     for(;n>0;)
     {
       a=n%10;
       sum=sum+pow(a,b);
       n=n/10;
     }
     if(sum==n)
     {
       cout<<"IT IS AN ARMSTRONG NUMBER...";
       getch();
     }
     else
     {
       cout<<"IT IS NOT AN ARMSTRONG NUMBER...";
       getch();
     }
   }

File: java/Armstrong Number.java



//Armstrong Number

   class Armstrong 
   { 
     public static void main(String args[]) 
     { 
       int num = Integer.parseInt(args[0]); 
       int n = num; //use to check at last time 
       int check=0,remainder; 
       while(num > 0) 
       { 
         remainder = num % 10; 
         check = check + (int)Math.pow(remainder,3); 
         num = num / 10; 
       } 
       if(check == n) 
         System.out.println(n+" is an Armstrong Number"); 
       else 
         System.out.println(n+" is not a Armstrong Number"); 
     } 
   } 
